### PR TITLE
Exposed filter states and added ability to not toggle the filter window display on applying filters.

### PR DESCRIPTION
**Pull request by umamialex**

### CODE REVIEW

File: ampersand-chart.js [hunk 1]
```diff
@@ -467,9 +473,11 @@
           }
         }]
       });
-      filterTrackerState.onApply = function(props) { 
+      filterTrackerState.onApply = function(props, options) { 
         this.model.filterOnApply(props);
-        this.toggleFilterWindow();
+        if (options.doNotToggle !== true) {
```
Comment (nubs) [inline ampersand-chart.js:478]: since it's boolean, why not `if (!options.doNotToggle) {`

File: ampersand-chart.js [hunk 2]
```diff
@@ -30,6 +30,12 @@
       searchSelectFilter: 'function',
       filterOnApply: [ 'function', false, _.constant(_.noop) ],
 
+      // Filter State
+      timeRangeState: 'object',
```
Comment (nubs) [inline ampersand-chart.js]: can these by `state` instead of `object`

File: ampersand-chart.js (hunk 1 above)
Comment (umamialex) [inline ampersand-chart.js:478]: It's actually mostly undefined and I don't like trusting JS's type coercion. /giphy paranoia

**Comment by umamialex** [comment]
Reupped.

**Pull request merged**